src/hotspot/share/interpreter/interpreterRuntime.cpp

```diff
@@ -735,6 +735,15 @@ void InterpreterRuntime::resolve_get_put(JavaThread* current, Bytecodes::Code by
 
 //%note monitor_1
 JRT_ENTRY_NO_ASYNC(void, InterpreterRuntime::monitorenter(JavaThread* current, BasicObjectLock* elem))
+  if (!UseHeavyMonitors && UseFastLocking) {
+    // This is a hack to get around the limitation of registers in x86_32. We really
+    // send an oopDesc* instead of a BasicObjectLock*.
```

Comment:
I don't understand what this is referring to.  Trying to avoid passing an extra argument?

Reply:
I updated the comment in an earlier commit to say:
   // TODO: We accept elem as void* to workaround a limitation of registers in x86_32. Interpreter
    // code is really sending an oopDesc* here.
    // The problem is that we would need to preserve the register that holds the BasicObjectLock,
    // but we are using that register to hold the thread. We don't have enough registers to
    // also keep the BasicObjectLock, but we don't really need it anyway, we only need
    // the object. See also InterpreterMacroAssembler::lock_object().
    // As soon as traditional stack-locking goes away we can change elem to be oopDesc*
    // (also in monitorexit below).
I hope that is clearer.